Bug 1450045 - Convert Logviewer to ReactJS

camd opened the pull request:
This sits on top of the move of ``LogoMenu`` to a shared component, so the first commit can be ignored.  I made a few design decisions here, so please feel free to chime in where you'd like changes.

The "Result: testfailed" looks a little brown to me on the now-dark nav bar.  Oh, and the navbar is now dark to match the other apps.

camd commented:
Oops, looks like I broke one of the Selenium tests.  I'll fix that tomorrow.  Feel free to un-review, if you like.  Though I doubt it'll be a big change to fix.

camd commented:
OK, Tests are fixed now.  :)  Ready for review when you are.  Thanks!!

camd commented:
OK, actually...  I'm going to disable the lint exceptions and go fix them in here as much as possible.  So, again, feel free to hold off reviewing, if you like.  Sorry for the yo-yo...

camd commented:
Linting errors fixed in the new files, but wasn't able to permanently enable any rules at this time.

sarah-clements commented on ui/log-view/logviewer.css:
```diff
@@ -0,0 +1,64 @@
+}
+
+.result-status {
+    color: lightgrey;
```
There's a `.lightgray` class in the `treeherder-global.css` file you could use instead.

camd replied:
Good call.  Thanks.  :)

camd commented on .neutrinorc.js:
```diff
@@ -13,8 +13,9 @@ module.exports = {
       logviewer: {
-        entry: 'entry-logviewer.js',
-        template: 'ui/logviewer.html',
+        entry: 'log-view/index.jsx',
```
Sure, wfm.  :)

camd commented:
@edmorley I think I've addressed all your issues.  Thanks for finding those bugs!

edmorley commented on package.json:
```diff
@@ -25,7 +25,6 @@
     "angular": "1.7.5",
     "angular-clipboard": "1.6.2",
     "angular-local-storage": "0.7.1",
-    "angular-sanitize": "1.7.5",
```
Could you update `yarn.lock` for this change?

camd commented:
@edmorley I fixed the toggling of a line.  Ends up there is an odd thing where setting ``highlight`` with ``state`` on the LazyLog works fine with numbers, but setting it to ``null`` does not clear it.  You have to set it to ``-1`` to clear it.  I submitted an issue upstream: https://github.com/mozilla-frontend-infra/react-lazylog/issues/22.

Also, for the "scroll to the bottom" I added a margin to the container.  On a mac, the scrollbars hide themselves after a few seconds, so I'm hoping this margin works.  :)

edmorley commented:
Those changes look great - thank you! I rebased on master (to resolve a `yarn.lock` conflict with a Renovate PR merge just now), added a comment linking to the react-lazylog issue, and lowered `maxAssetSize` to reflect the new slimmer index/perf entrypoints (due to more shared code with the JobInfo component, and I think also one less thing using angular so therefore less contention for the shared chunks).

I'll merge this now so I can rebase other eslint branches I have locally on top of it :-)
